turbopy/diagnostics.py

```diff
@@ -371,6 +380,8 @@ def __init__(self, owner: Simulation, input_data: dict):
         self.field_was_found = False
 
         self.outputter = None
+        self.handler = None
+        self.interval = self._input_data.get('write_interval', None)
 
     def check_step(self):
```

Comment:
This class should be changed to use an interval handler to deal with the `dump_interval` parameter. Basically, there should be two intervals... one that says how often `diagnose` gets called, and one that says how often the data gets written to file.

Another way to look at this is to realized that having `check_step` code here is now duplicating the functionality that the IntervalHandler provides.